### PR TITLE
boards: arm: cxd56xx: enable basic snapshot camera example

### PR DESCRIPTION
**Pull request by jerpelea**
## Summary
This camera example grabs a specified number of pictures
and saves them on the available storage

## Impact
Spresense board 

## Testing
built for Spresense board

### CODE REVIEW

**Comment by jerpelea** [comment]
@masayuki2009

**Comment by MasayukiIshikawa** [comment]
@jerpelea 
Do you know why the build error happened?

**Comment by masayuki2009** [comment]
@jerpelea
Do you know why the build error happened?

**Comment by Ouss4** [comment]
> @jerpelea
> Do you know why the build error happened?

spresense/example_camera defconfig needs to be updated as well, with ./tools/refresh.sh

**Comment by jerpelea** [comment]
@Ouss4 the config was updated with savedefconfig

**Comment by jerpelea** [comment]
I suspect that if fails because the example app is not merged in the app git

**Comment by Ouss4** [comment]
> I suspect that if fails because the example app is not merged in the app git

Yes, that could be the issue, CONFIG_EXAMPLES_CAMERA still doesn't exist until apps#314 is merged.

**Comment by masayuki2009** [comment]
> I suspect that if fails because the example app is not merged in the app git
> 
> [apache/incubator-nuttx-apps#314](https://github.com/apache/incubator-nuttx-apps/pull/314)

I've just merged the PR.

**Comment by Ouss4** [comment]
I restarted the checks multiple times but the issue persists.  I pulled this change locally, it builds cleanly.  Also my refresh.sh does not produce a different `defconfig`.
But here CONFIG_EXAMPLE_CAMERA is not recognized as new option and removed from this `defconfig` when refresh.sh does `oldconfig`.
And, yes the checks are pulling the latest apps/.

**Comment by btashton** [comment]
I grabbed the source bundle from the CI and saw a bunch of these errors when running the build arm-01.dat test file. Maybe we can rebase this and see if it passes now that some of the other config flags have been fixed?  This last build ran while there were still issues with apps and `CONFIG_SYSTEM_NSH_CXXINITIALIZE`

**Comment by Ouss4** [comment]
@jerpelea @btashton What to do here?  @jerpelea Would you like to rebase and give it another go?
The PR's change is safe, the rebase would just check if everything is working as supposed to.

**Comment by Ouss4** [comment]
FYI, I merged this one to see how it would do once pushed to master and especially in the nightly build.  So far not good, the same result in master (actually expected), let's see how things will go at night.

**Comment by acassis** [comment]
HEAD detached at pull/1388/merge
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   boards/arm/cxd56xx/spresense/configs/example_camera/defconfig

no changes added to commit (use "git add" and/or "git commit -a")
  Configuring...
  Copy files
  Select CONFIG_HOST_LINUX=y
  Refreshing...
  Disabling CONFIG_ARMV7A_TOOLCHAIN_GNU_EABIL
  Enabling CONFIG_ARMV7M_TOOLCHAIN_GNU_EABIL
  Building NuttX...
  Normalize sabre-6quad/nsh
HEAD detached at pull/1388/merge
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   boards/arm/cxd56xx/spresense/configs/example_camera/defconfig

no changes added to commit (use "git add" and/or "git commit -a")
====================================================================================

File: boards/arm/cxd56xx/spresense/configs/example_camera/defconfig [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+CONFIG_SMARTFS_MULTI_ROOT_DIRS=y
+CONFIG_SPECIFIC_DRIVERS=y
+CONFIG_SPI=y
+CONFIG_SPRESENSE_EXTENSION=y
+CONFIG_START_DAY=6
+CONFIG_START_MONTH=12
+CONFIG_START_YEAR=2011
+CONFIG_SYSTEM_CLE=y
+CONFIG_SYSTEM_NSH=y
+CONFIG_SYSTEM_NSH_CXXINITIALIZE=y
```
Comment (xiaoxiang781216) [inline boards/arm/cxd56xx/spresense/configs/example_camera/defconfig:84]: don't exist any more. Please see this PR:
https://github.com/apache/incubator-nuttx-apps/pull/316

**Comment by xiaoxiang781216** [comment]
Fixed here: #1389

**Comment by masayuki2009** [comment]
>don't exist any more. Please see this PR:
>apache/incubator-nuttx-apps#316

I think so too.

```
$ make savedefconfig
$ diff -c defconfig ./boards/arm/cxd56xx/spresense/configs/example_camera/defconfig 
*** defconfig	2020-07-09 11:28:50.523639892 +0900
--- ./boards/arm/cxd56xx/spresense/configs/example_camera/defconfig	2020-07-09 08:00:04.022527937 +0900
***************
*** 81,86 ****
--- 81,87 ----
  CONFIG_START_YEAR=2011
  CONFIG_SYSTEM_CLE=y
  CONFIG_SYSTEM_NSH=y
+ CONFIG_SYSTEM_NSH_CXXINITIALIZE=y
  CONFIG_SYSTEM_USBMSC=y
  CONFIG_UART1_SERIAL_CONSOLE=y
  CONFIG_USBDEV=y
```

I will send a new PR to fix this issue.

**Comment by jerpelea** [comment]
@masayuki2009  @xiaoxiang781216 thanks for looking into it 
I wonder why savedefconfig did not fix it

**Comment by xiaoxiang781216** [comment]
Because CONFIG_NSH_CXXINITIALIZE is justed remove a few day ago, your local copy mayn't contain this patch:
https://github.com/apache/incubator-nuttx-apps/pull/316

**Comment by jerpelea** [comment]
@xiaoxiang781216 thanks

**Pull request merged**